lightly/active_learning/scorers/keypoint_detection.py

```diff
@@ -0,0 +1,91 @@
+from typing import List, Dict
+
+import numpy as np
+
+from lightly.active_learning.scorers import Scorer
+from lightly.active_learning.utils.keypoint_detection_output import \
+    KeypointDetectionOutput, KeypointDetection
+
+
+def _least_confidence(
+        model_output: List[KeypointDetectionOutput]) -> np.ndarray:
+    """Score which prefers samples with low confidence score.
+    
+    The confidence score per image is 1 minus the mean confidence
+    score of all its keypoints.
+
+    Args:
+        model_output:
+            Predictions of the model of length N.
+
+    Returns:
+        Numpy array of length N with the computed scores.
+
+    """
+    scores = []
+    for keypoint_detection_output in model_output:
+        confidences_this_detection = []
+        for keypoint_detection in keypoint_detection_output.keypoint_detections:
+            confidences = keypoint_detection.get_confidences()
+            if len(confidences) > 0:
+                conf = np.mean(confidences)
+                confidences_this_detection.append(conf)
+        if len(confidences_this_detection) > 0:
+            score = 1. - np.mean(confidences_this_detection)
+            scores.append(score)
+    return np.asarray(scores)
+
+
+class ScorerKeypointDetection(Scorer):
+    """Class to compute active learning scores from the model_output of a keypoint detection task.
+
+    Currently supports the following scorers:
+
+        `least_confidence`:
```

Comment:
We should at least support `uncertainty_margin`

Reply:
We don't have class distributions, thus `uncertainty_margin` and `uncertainty_entropy` don't make sense.

Reply:
Thus also "least_confidence" is the wrong name, I renamed it to "mean_uncertainty" as it is `1 - mean(confidences over all keypoints)`.